backend/sources/viber/connector/src/main/java/co/airy/core/sources/viber/services/Api.java

```diff
@@ -81,4 +92,33 @@ public void setWebhook(String webhookUrl) throws Exception {
     public void removeWebhook() throws Exception {
         getApiResponse("/set_webhook", objectMapper.writeValueAsString(Map.of("url", "")));
     }
+
+
+    @Override
+    public void onApplicationEvent(ApplicationReadyEvent event) {
+        restTemplate = restTemplateBuilder
+                .errorHandler(new ResponseErrorHandler() {
```

Comment:
Interesting anonymous classes 🤔

Reply:
Yeah. By the way, there is yet another library which we used in the past and that could allow us to write declarative, simpler and thus more maintainable REST clients: https://github.com/OpenFeign/feign 

The best case though is that we rely on the new [source api](https://airy.co/docs/core/api/source) and write those source clients in a different language (go perhaps? ;) )

Reply:
> Yeah. By the way, there is yet another library which we used in the past and that could allow us to write declarative, simpler and thus more maintainable REST clients: https://github.com/OpenFeign/feign

I will have a look thanks

> The best case though is that we rely on the new source api and write those source clients in a different language (go perhaps? ;) )

That will be a great Idea. We can discuss that further ![](https://gist.githubusercontent.com/GeekTree0101/05d338bb59109fc71871711c6fa49377/raw/3ff868ffcf2f84d419c392667335fe7e9f1bf155/dancing-gopher.gif)